src/main/java/io/vertx/ext/sync/SyncVerticle.java

```diff
@@ -1,83 +1,73 @@
 package io.vertx.ext.sync;
 
-import co.paralleluniverse.fibers.*;
-import io.vertx.core.*;
+import co.paralleluniverse.fibers.FiberScheduler;
+import co.paralleluniverse.fibers.Suspendable;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
 
 /**
+ * A `Verticle` which runs its `start` and `stop` methods using fibers.
+ *
+ * You should subclass this class instead of `AbstractVerticle` to create any verticles that use vertx-sync.
  *
  * @author <a href="http://tfox.org">Tim Fox</a>
  */
 public abstract class SyncVerticle extends AbstractVerticle {
 
   protected FiberScheduler instanceScheduler;
-  protected Thread eventLoop;
 
   @Override
   public void start(Future<Void> startFuture) throws Exception {
-    if (context.isWorkerContext()) {
-      throw new IllegalStateException("Vert.x sync can only be used with standard verticles");
-    }
-    eventLoop = Thread.currentThread();
-    instanceScheduler = new FiberExecutorScheduler("mine", command -> {
-      if (Thread.currentThread() != eventLoop) {
-        context.runOnContext(v -> command.run());
-      } else {
-        // Just run directly
-        command.run();
-      }
-    });
-    new Fiber<Void>(instanceScheduler) {
+    instanceScheduler = Sync.getContextScheduler();
+    Sync.runOnFiber(instanceScheduler, new Runnable() {
+      // TODO - for some reason this does not work if we use a lambda instead of anonymous class
```

Comment:
That's because you can't annotate lambdas... which is why `FiberUtil` [uses `SuspendableRunnable`](https://github.com/puniverse/quasar/blob/master/quasar-core/src/main/java/co/paralleluniverse/fibers/FiberUtil.java#L94), which is a functional interface already declared to pause, and therefore supports lambdas.